### ISSUE TITLE
[SUGGESTION] Allow categories to be excluded from breeding

### ISSUE THREAD

**Issue by shallotmama** [pokeclicker/pokeclicker#2451]
**Current Behavior**

Selecting a category from the breeding menu filter includes only pokemon in that category.

**Improved Behavior**

The option to _exclude_ instead would be useful. It would also potentially be useful to extend this to the Type filters (for example, to round out the party's attack types without restricting it to one type).

**Reason for change**

It would cover some annoying scenarios, such as the following:
I want to keep breeding as I'm farming coins/dungeon tokens/exp, but if I breed my highest-attack mons constantly, it can make it so I can't one-shot enemies in the area I want to farm. I want to be able to exclude specific pokemon so my attack doesn't drop so drastically when I'm breeding.

**Detailed Description**

I imagine a reversal of the filter would be relatively straightforward, with a checkbox next to the category options to toggle whether it's included or excluded. If we wanted to go even deeper, a custom category combining option could be added to allow for chaining logic like "Include Category A and B, except those also in Category C." This would go well with the proposition in #2399. I'm in favor of the latter, but the former is a quicker add.

**Screenshots:**

**Comment by Somerandomguy70** [comment]
I think adding a 'Shiny Last' button under the 'Shiny Status' category would also be a good improvement. That way, people who are breeding and trying to shiny hunt at the same time will have the non-shiny mons placed first in the hatchery queue, followed by their shiny mons. Would be nice for people who have a lot of shiny mons, and would otherwise need to manually go through, find, and breed the non shiny mons first, followed by shiny mons. Just a small QOL I think.

**Comment by Symi001** [comment]
> I think adding a 'Shiny Last' button under the 'Shiny Status' category would also be a good improvement. That way, people who are breeding and trying to shiny hunt at the same time will have the non-shiny mons placed first in the hatchery queue, followed by their shiny mons. Would be nice for people who have a lot of shiny mons, and would otherwise need to manually go through, find, and breed the non shiny mons first, followed by shiny mons. Just a small QOL I think.

You can sort by Shiny, which does exactly what you're describing, and this is not linked to categories.

**Comment by Somerandomguy70** [comment]
> > I think adding a 'Shiny Last' button under the 'Shiny Status' category would also be a good improvement. That way, people who are breeding and trying to shiny hunt at the same time will have the non-shiny mons placed first in the hatchery queue, followed by their shiny mons. Would be nice for people who have a lot of shiny mons, and would otherwise need to manually go through, find, and breed the non shiny mons first, followed by shiny mons. Just a small QOL I think.
> 
> You can sort by Shiny, which does exactly what you're describing, and this is not linked to categories.

Not exactly, sorting by shiny hides all non-shiny mons, sorting by not shiny hides all shiny mons. My suggestion would not do this, it would merely place the shiny mons after the regular mons, that way people could easily breed through their non shiny mons first, and then move onto the shiny mons immediately, like I said, a small QOL. But I understand this may not be the best place to talk about this, apologies.

**Comment by PixLSteam** [comment]
> Not exactly, sorting by shiny hides all non-shiny mons, sorting by not shiny hides all shiny mons. My suggestion would not do this, it would merely place the shiny mons after the regular mons, that way people could easily breed through their non shiny mons first, and then move onto the shiny mons immediately, like I said, a small QOL. But I understand this may not be the best place to talk about this, apologies.

You're confusing "sorting" and "filtering". There *is* a shiny *sorting* option. But please don't hijack other suggestions.

**Comment by Somerandomguy70** [comment]
> > Not exactly, sorting by shiny hides all non-shiny mons, sorting by not shiny hides all shiny mons. My suggestion would not do this, it would merely place the shiny mons after the regular mons, that way people could easily breed through their non shiny mons first, and then move onto the shiny mons immediately, like I said, a small QOL. But I understand this may not be the best place to talk about this, apologies.
> 
> You're confusing "sorting" and "filtering". There _is_ a shiny _sorting_ option. But please don't hijack other suggestions.

Oh wow, I feel dumb... tysm! I'll delete my posts.
